### PR TITLE
UI Changes and admin rights

### PR DESCRIPTION
**Pull request by Reznal**

### CODE REVIEW

File: app/controllers/welcome_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class WelcomeController < ApplicationController
+before_action:logged_in?
+skip_before_action:logged_in? , except: [:index]
```
Comment (Kostronor) [inline app/controllers/welcome_controller.rb]: why the : why not make a space instead?
`skip_before_action logged_in?, except: [:index]`

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -27,4 +27,9 @@ body {
 
 .main {
   flex: 1 0 auto;
-}
\ No newline at end of file
+}
+
```
Comment (Reznal) [inline app/assets/stylesheets/application.scss]: Complaining about using an id tag

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -13,3 +13,7 @@ def ensure_player
     end
   end
 end
+
+def logged_in?
+  redirect_to job_boards_path if current_user
+end
```
Comment (Reznal) [inline app/controllers/application_controller.rb]: Have added final newline, not detecting.

File: app/assets/stylesheets/application.scss (hunk 2 above)
Comment (Reznal) [inline app/assets/stylesheets/application.scss]: Complaining about id tags, we need and id tag where this is used.

File: app/controllers/players_controller.rb [hunk 4]
```diff
@@ -70,7 +83,7 @@ def no_duplicate_player
 
     # Never trust parameters from the scary internet, only allow the white list through.
     def player_params
-      params.require(:player).permit(:name)
+      params.require(:player).permit(:name, :admin)
```
Comment (Kostronor) [inline app/controllers/players_controller.rb]: Unsafe parameter admin, we don't want that open

File: app/controllers/players_controller.rb [hunk 5]
```diff
@@ -26,11 +31,19 @@ def new
   # This will always show your own player
   # and #update will always update your own player
   def edit
+    @current_player = Player.find(params[:id])
```
Comment (Kostronor) [inline app/controllers/players_controller.rb]: I saw the before_action we should update the method comment to reflect that change

File: app/controllers/players_controller.rb [hunk 6]
```diff
@@ -2,11 +2,16 @@ class PlayersController < ApplicationController
   before_action :ensure_player, except: [:new, :create]
   before_action :authenticate_user!, only: [:new, :create]
   before_action :no_duplicate_player, only: [:new, :create]
+  before_action :editing_self?, only: [:edit]
```
Comment (Kostronor) [inline app/controllers/players_controller.rb]: maybe use this for update too?

File: app/controllers/welcome_controller.rb [hunk 7]
```diff
@@ -1,4 +1,5 @@
 class WelcomeController < ApplicationController
+  before_action:logged_in?, except: [:index]
```
Comment (Kostronor) [inline app/controllers/welcome_controller.rb]: does this block not logged in users?
if yes it needs anonymous included
if no, why exclude index?

File: app/views/players/_form.html-test.erb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+-<%= form_for(player) do |f| %>
```
Comment (Kostronor) [inline app/views/players/_form.html-test.erb]: why rename this?

File: app/controllers/players_controller.rb [hunk 9]
```diff
@@ -26,11 +31,19 @@ def new
   # This will always show your own player
   # and #update will always update your own player
   def edit
+    @current_player = Player.find(params[:id])
   end
 
   # do nothing for now
   # changing player name is not supported
   def update
+    @current_player = Player.find(params[:id])
```
Comment (Kostronor) [inline app/controllers/players_controller.rb]: make this a before_action

File: app/controllers/players_controller.rb [hunk 10]
```diff
@@ -2,11 +2,16 @@ class PlayersController < ApplicationController
   before_action :ensure_player, except: [:new, :create]
   before_action :authenticate_user!, only: [:new, :create]
   before_action :no_duplicate_player, only: [:new, :create]
+  before_action :editing_self?, only: [:edit]
+
+  def editing_self?
```
Comment (Kostronor) [inline app/controllers/players_controller.rb]: make this private

File: app/views/players/_form.html-test.erb (hunk 8 above)
Comment (Reznal) [inline app/views/players/_form.html-test.erb]: Was supposed to be another file, not sure what happened, have reverted

File: app/controllers/players_controller.rb (hunk 6 above)
Comment (Reznal) [inline app/controllers/players_controller.rb]: Done

File: app/controllers/players_controller.rb (hunk 10 above)
Comment (Reznal) [inline app/controllers/players_controller.rb]: Where would we put it so an admin can update another players profile to be admin

File: app/controllers/players_controller.rb (hunk 4 above)
Comment (Reznal) [inline app/controllers/players_controller.rb]: Where would we put it so an admin can update another players profile to be admin

File: app/controllers/welcome_controller.rb (hunk 7 above)
Comment (Reznal) [inline app/controllers/welcome_controller.rb]: Stopped a logged in used from seeing index, Doesn't feel necessary currently for someone logged in to see it, but I can change it up to be more like the old one?

**Pull request merged**